cmd/kubeadm/app/cmd/init.go

```diff
@@ -150,6 +157,8 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 
 	AddInitConfigFlags(cmd.PersistentFlags(), externalcfg, &featureGatesString)
 	AddInitOtherFlags(cmd.PersistentFlags(), &cfgPath, &skipPreFlight, &skipTokenPrint, &dryRun, &ignorePreflightErrors)
+	bto.AddTokenFlag(cmd.PersistentFlags())
```

Comment:
What about changing `Add` methods in order to accept in input also `externalCfg`; doing this we could avoid to change following lines above:
```go
err = bto.ApplyTo(externalcfg)
kubeadmutil.CheckErr(err)
```
And help in keeping the flag logic isolated from the command logic

Reply:
The ApplyTo is required as it can only run at runtime, when the command is executed.
It doesn't make sense to run this code at set-up time.